src/Compilers/CSharp/Test/Semantic/Semantics/NullableReferenceTypesTests.cs

```diff
@@ -32552,6 +33136,110 @@ static void M1<T>(T t1)
             comp2.VerifyDiagnostics();
         }
 
+        [Fact, WorkItem(36073, "https://github.com/dotnet/roslyn/issues/36073")]
+        public void NotNull_Property_InitializedInConstructor()
+        {
+            var lib_cs =
+@"using System.Diagnostics.CodeAnalysis;
+public class COpen<TOpen>
+{
+    [NotNull]public TOpen P1 { get; set; }
+    COpen()
+    {
+        P1 = default; // 1
+    }
+}
+public class CClass<TClass> where TClass : class
+{
+    [NotNull]public TClass P2 { get; set; }
+    [NotNull]public TClass? P3 { get; set; }
+    CClass()
+    {
+        P2 = null; // 2
+        P3 = null;
+    }
+}
+public class CStruct<TStruct> where TStruct : struct
+{
+    [NotNull]public TStruct P4 { get; set; }
+    [NotNull]public TStruct? P5 { get; set; }
+    CStruct()
+    {
+        P4 = default;
+        P5 = null;
+    }
+}";
+            var comp = CreateNullableCompilation(new[] { lib_cs, NotNullAttributeDefinition });
+            comp.VerifyDiagnostics(
+                // (7,14): warning CS8601: Possible null reference assignment.
+                //         P1 = default; // 1
+                Diagnostic(ErrorCode.WRN_NullReferenceAssignment, "default").WithLocation(7, 14),
+                // (16,14): warning CS8625: Cannot convert null literal to non-nullable reference type.
+                //         P2 = null; // 2
+                Diagnostic(ErrorCode.WRN_NullAsNonNullable, "null").WithLocation(16, 14)
+                );
+        }
+
+        [Fact, WorkItem(36073, "https://github.com/dotnet/roslyn/issues/36073")]
+        public void NotNull_Property_InitializedInConstructor_WithValues()
+        {
+            var lib_cs =
+@"using System.Diagnostics.CodeAnalysis;
+public class COpen<TOpen> where TOpen : new()
+{
+    [NotNull]public TOpen P1 { get; set; }
```

Comment:
Is this just additional coverage, or did your change affect behavior of these scenarios? It doesn't seem like enforcing unconditional attributes inside the members using the attributes affects this?

Reply:
For a while, this PR include analyzing the body of the auto-prop accessors. So those tests were useful.
Since I added the tests, I'd rather keep them even though they are less useful now :-)

---
In reply to: [368177028](https://github.com/dotnet/roslyn/pull/40789#discussion_r368177028) [](ancestors = 368177028)